velox/expression/tests/ArrayProxyTest.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glog/logging.h"
+#include "gtest/gtest.h"
+#include "velox/expression/VectorUdfTypeSystem.h"
+#include "velox/functions/Udf.h"
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+namespace facebook::velox {
+namespace {
+// Function that creates array with values 0...n-1.
+// Uses all possible functions in the array proxy interface.
+template <typename T>
+struct Func {
+  template <typename TOut>
+  bool call(TOut& out, const int64_t& n) {
```

Comment:
any reason you are not using out_arg<...> here?

Reply:
no I can do that, 
The only reason is when I replace ArrayProxyT with Array at the end i wont have to change those.

Reply:
https://github.com/facebookincubator/velox/pull/787

same thing was done here. 

The compiler wont be able to deduce types of the call function.

Reply:
This is an unfortunate limitation. Should we describe it in the documentation?

Reply:
@pedroerp